### PR TITLE
Add parsing for UDT

### PR DESCRIPTION
**Pull request by akva2**
This adds parsing support for 1D UDTs. Sits on top of https://github.com/OPM/opm-common/pull/3725

### CODE REVIEW

File: src/opm/input/eclipse/Schedule/KeywordHandlers.cpp [hunk 1]
```diff
@@ -1174,6 +1175,63 @@ File {} line {}.)", wname, location.keyword, location.filename, location.lineno)
...
+        } else if (interp_type == "LL") {
+            type = UDT::InterpolationType::LinearExtrapolate;
+        } else {
+            throw OpmInputError(fmt::format("Unknown UDT interpolation type {}", interp_type),
+                                handlerContext.keyword.location());
+        }
+        const auto x_vals = points.getItem<ParserKeywords::UDT::INTERPOLATION_POINTS>().getData<double>();
+
+        if (!std::is_sorted(x_vals.begin(), x_vals.end())) {
+            throw OpmInputError("UDT: Interpolation points should be given in ascending order",
```
Comment (blattms) [inline src/opm/input/eclipse/Schedule/KeywordHandlers.cpp:1210]: "should" seems to weak. Maybe use "need to"?

File: src/opm/input/eclipse/Schedule/KeywordHandlers.cpp [hunk 2]
```diff
@@ -1174,6 +1175,63 @@ File {} line {}.)", wname, location.keyword, location.filename, location.lineno)
...
+                                handlerContext.keyword.location());
+        }
+
+        if (auto it = std::adjacent_find(x_vals.begin(), x_vals.end()); it != x_vals.end()) {
+            throw OpmInputError(fmt::format("UDT: Interpolation points should be unique: "
+                                            "found duplicate for {}", *it),
```
Comment (blattms) [inline src/opm/input/eclipse/Schedule/KeywordHandlers.cpp:1216]: Same here.

**Review by blattms — COMMENTED** [review]
Just some nitpicking on the error messages :sweat_smile: 
Should we also test whether the data in the table is correct after parsing?

File: src/opm/input/eclipse/Schedule/KeywordHandlers.cpp (hunk 1 above)
Comment (akva2) [reply src/opm/input/eclipse/Schedule/KeywordHandlers.cpp]: changed to need to

File: src/opm/input/eclipse/Schedule/KeywordHandlers.cpp (hunk 2 above)
Comment (akva2) [reply src/opm/input/eclipse/Schedule/KeywordHandlers.cpp:1216]: changed to need to

**Comment by akva2** [comment]
jenkins build this please

**Pull request merged**